lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp

```diff
@@ -0,0 +1,261 @@
+      smf_context->gtp_tunnel_id.upf_gtp_teid, 4);
+  amf_pdu_ses_setup_transfer_req.up_transport_layer_info.gtp_tnl
+      .endpoint_ip_address =
+      blk2bstr(&smf_context->gtp_tunnel_id.upf_gtp_teid_ip_addr, 4);
+  amf_pdu_ses_setup_transfer_req.pdu_ip_type.pdn_type = IPv4;
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
+      .qos_flow_identifier =
+      smf_context->pdu_resource_setup_req
+          .pdu_session_resource_setup_request_transfer
+          .qos_flow_setup_request_list.qos_flow_req_item.qos_flow_identifier;
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
+      .qos_flow_level_qos_param.qos_characteristic.non_dynamic_5QI_desc.fiveQI =
+      smf_context->pdu_resource_setup_req
+          .pdu_session_resource_setup_request_transfer
+          .qos_flow_setup_request_list.qos_flow_req_item
+          .qos_flow_level_qos_param.qos_characteristic.non_dynamic_5QI_desc
+          .fiveQI;
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
+      .qos_flow_level_qos_param.alloc_reten_priority.priority_level =
+      smf_context->pdu_resource_setup_req
+          .pdu_session_resource_setup_request_transfer
+          .qos_flow_setup_request_list.qos_flow_req_item
+          .qos_flow_level_qos_param.alloc_reten_priority.priority_level;
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
+      .qos_flow_level_qos_param.alloc_reten_priority.pre_emption_cap =
+
+      // MAY_TRIGGER_PRE_EMPTION;
+      smf_context->pdu_resource_setup_req
+          .pdu_session_resource_setup_request_transfer
+          .qos_flow_setup_request_list.qos_flow_req_item
+          .qos_flow_level_qos_param.alloc_reten_priority.pre_emption_cap;
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
+      .qos_flow_level_qos_param.alloc_reten_priority.pre_emption_vul =
+
+      // PRE_EMPTABLE;
+      smf_context->pdu_resource_setup_req
+          .pdu_session_resource_setup_request_transfer
+          .qos_flow_setup_request_list.qos_flow_req_item
+          .qos_flow_level_qos_param.alloc_reten_priority.pre_emption_vul;
+
+  // Adding respective header to amf_pdu_ses_setup_transfer_request
+  // Convert amf_pdu_ses_setup_transfer_req to bstring and assign to message_p
+  OAILOG_DEBUG(
+      LOG_AMF_APP,
+      "Converting pdu_session_resource_setup_request_transfer_t to bstring \n");
+  ngap_pdu_ses_setup_req->pduSessionResource_setup_list.item[0]
+      .PDU_Session_Resource_Setup_Request_Transfer =
+      amf_pdu_ses_setup_transfer_req;
+
+  // Send message to NGAP task
+  send_msg_to_task(&amf_app_task_zmq_ctx, TASK_NGAP, message_p);
+
+  return RETURNok;
+}
+
+/* Resourece release request to gNB through NGAP */
+int pdu_session_resource_release_request(
+    ue_m5gmm_context_s* ue_context, amf_ue_ngap_id_t amf_ue_ngap_id) {
+  if (!ue_context) {
+    // TODO
```

Comment:
again don't leave this dangling. if it is nullptr, log error message and return error code.

Reply:
fixed